packages/store/addon/-private/system/weak-cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DEBUG } from '@glimmer/env';
+
+import { addSymbol, symbol } from '../ts-interfaces/utils/symbol';
+
+interface WeakCache<K extends object, V> {
```

Comment:
Also, why do we need a separate interface for the concrete class authored beneath?

Reply:
I'll leave a comment, this is a pattern in our toolbox we have used a few places where we utilize typescripts declaration merging to our advantage. Basically, it prevents typescript from inserting lots of `void 0` initializers for things that may not be present. This is especially valuable for properties that are only present in DEBUG where typing the concrete class with them would include those fields and their initialization even in production.